core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1987,7 +1987,11 @@ else if (sourceNode instanceof SampleNode) {
                 throw e;
             }
             catch (RuntimeException e) {
-                throw new TrinoException(COMPILER_ERROR, "Compiler failed", e);
+                throw new TrinoException(
+                        COMPILER_ERROR,
+                        "Compiler failed. Possible reasons include: the query may have too many or too complex expressions, " +
+                                "or the underlying tables may have too many columns",
```

Comment:
Nice improvement. Can we go even further and do some classification of `e` and provide different exception messages depending on class of `e`. (no changes requested)

Reply:
I don't think we have enough context to do this. I would rather leave this for a follow-up (if at all)